### PR TITLE
[PD-2892] initial commit of cbo ui for knowledge transfer to Congress…

### PR DESCRIPTION
**Pull request by ghost**
…man Carlos Danger

Will video chat with Representative Danger for knowledge transfer.

### CODE REVIEW

**Comment by koonsies** [comment]
I will be taking over this project. I will leave this up for visibility, but will not attempt to get it merged until I get a better handle on the code.

File: gulp/package.json [hunk 1]
```diff
@@ -66,8 +67,12 @@
...
     "sortablejs": "^1.4.2",
     "tinycolor2": "~1.3.0",
+    "toastr": "2.1.2",
     "warning": "^2.1.0",
     "webpack": "^1.12.9",
     "webpack-dev-server": "^1.14.1"
-  }
+  },
+   "devDependencies": {
+     "redux-immutable-state-invariant": "1.2.3"
```
Comment (darrint) [inline gulp/package.json]: This needs to move to dependencies.

File: gulp/src/cbo/store/configureStore.dev.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import {createStore, applyMiddleware} from 'redux';
+import rootReducer from '../reducers';
+import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
+import thunk from 'redux-thunk';
+
+export default function configureStore(initialState) {
+  return createStore(
+    rootReducer,
+    initialState,
+    applyMiddleware(thunk, reduxImmutableStateInvariant())
```
Comment (darrint) [inline gulp/src/cbo/store/configureStore.dev.js]: This needs to be moved to the createStore function and only engaged in development, not production.

**Comment by darrint** [comment]
The npm-shrinkwrap file appears to have been modified without running npm clean-shrinkwrap. The diff shouldn't be that big.

File: gulp/package.json (hunk 1 above)
Comment (koonsies) [inline gulp/package.json]: I haven't researched this myself, but does is devDependencies only for packages required when the node settings are set to "development"?

Comment (koonsies) [inline gulp/package.json]: I know I've done --save and --save-dev (or something like that before). When would we use --save-dev?

File: gulp/src/cbo/store/configureStore.dev.js (hunk 2 above)
Comment (koonsies) [inline gulp/src/cbo/store/configureStore.dev.js]: They way he has it set up, configureStore is..

```
if (process.env.NODE_ENV === 'production') {
  module.exports = require('./configureStore.prod');
} else {
  module.exports = require('./configureStore.dev');
}
```

Is that not sufficient?

**Comment by koonsies** [comment]
I'm not seeing any references to the command npm clean-shrinkwrap.. Is that the right command? If so, what does it do?

File: gulp/package.json (hunk 1 above)
Comment (darrint) [inline gulp/package.json]: For now never. It's too hard to keep the shrinkwrap file straight with `devDependencies`.

File: gulp/src/cbo/store/configureStore.dev.js (hunk 2 above)
Comment (darrint) [inline gulp/src/cbo/store/configureStore.dev.js]: Where is that?

**Comment by darrint** [comment]
https://github.com/DirectEmployers/MyJobs/blob/quality-control/gulp/package.json#L7

You just run it.

**Comment by koonsies** [comment]
Closing pending further review

**Pull request closed without merging**